### PR TITLE
 bitw: init at unstable-2023-04-08 

### PR DESCRIPTION
**Pull request by lheckemann**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/vaultwarden.nix [hunk 1]
```diff
@@ -189,6 +198,12 @@ let
               f"bw --nointeraction --raw --session {key} list items | ${pkgs.jq}/bin/jq -r .[].login.password"
           )
           assert password.strip() == "${storedPassword}"
+
+      with subtest("get the password from the freedesktop secrets api"):
+          client.succeed("sudo -iu alice mkdir -p /home/alice/.config/bitw")
+          client.succeed("sudo -iu alice cp ${bitwConfig} /home/alice/.config/bitw/config")
+          client.succeed("sudo -iu alice env PASSWORD=${userPassword} bitw login")
+          # todo: start a proper session for alice, in which bitw will ideally be started through dbus activation when `secret-tool` is used.
```
Comment (lheckemann) [inline nixos/tests/vaultwarden.nix:206]: If anyone already knows how to do this, input is welcome.

File: nixos/tests/vaultwarden.nix [hunk 2]
```diff
@@ -156,7 +163,9 @@ let
 
       client = { pkgs, ... }:
         {
-          environment.systemPackages = [ pkgs.bitwarden-cli ];
+          imports = [ ./common/user-account.nix ];
+          environment.systemPackages = [ pkgs.bitwarden-cli pkgs.bitw pkgs.libsecret ];
```
Comment (SuperSandro2000) [inline nixos/tests/vaultwarden.nix:167]:
```suggestion
          environment.systemPackages = with pkgs; [ bitwarden-cli bitw libsecret ];
```

Comment (SuperSandro2000) [inline nixos/tests/vaultwarden.nix:167]:
```suggestion
          environment.systemPackages = with pkgs; [ bitwarden-cli bitw libsecret ];
```

File: pkgs/tools/security/bitw/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  # Requires connection to bitwarden.com.
+  doCheck = false;
+
+  src = fetchFromGitHub {
+    owner = "mvdan";
+    repo = "bitw";
+    rev = "b6162395999b253c7ddff7d3a40b380a4a373ec7";
+    sha256 = "0fglyw5gidybzyjlqi255yqrgdx1dp1a4jb5in4gfa9p87gd6nvm";
+  };
```
Comment (SuperSandro2000) [inline pkgs/tools/security/bitw/default.nix:16]:
```suggestion
{ fetchFromGitHub, buildGoModule, lib }:

buildGoModule rec {
  pname = "bitw";
  version = "unstable-2023-04-08";

  src = fetchFromGitHub {
    owner = "mvdan";
    repo = "bitw";
    rev = "b6162395999b253c7ddff7d3a40b380a4a373ec7";
    sha256 = "0fglyw5gidybzyjlqi255yqrgdx1dp1a4jb5in4gfa9p87gd6nvm";
  };

  vendorHash = "sha256-L6kHDZt0+QMC9BCBgh0CMyfD1lCb8ymq1sl4QqoCGH0=";

  # Requires connection to bitwarden.com.
  doCheck = false;
```

File: pkgs/tools/security/bitw/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/mvdan/bitw";
+    description = "Bitwarden client implementing the freedesktop secrets API";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ lheckemann ma27 hexa raitobezarius ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/bitw/default.nix:22]: Did you just copy maintainers from another package by chance?

Comment (mweinelt) [reply pkgs/tools/security/bitw/default.nix:22]: No, we talked about it on matrix.

File: pkgs/tools/security/bitw/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    rev = "b6162395999b253c7ddff7d3a40b380a4a373ec7";
+    sha256 = "0fglyw5gidybzyjlqi255yqrgdx1dp1a4jb5in4gfa9p87gd6nvm";
+  };
+
+  passthru.tests = [ nixosTests.vaultwarden ];
+  meta = with lib; {
+    homepage = "https://github.com/mvdan/bitw";
+    description = "Bitwarden client implementing the freedesktop secrets API";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ lheckemann ma27 hexa raitobezarius ];
```
Comment (Ma27) [inline pkgs/tools/security/bitw/default.nix:23]: I signed up for this when I was actively considering to switch to bitwarden anyways soonish. Recently I decided against switching my entire setup for it, so I won't have a use for this software.

Hence, I don't think it makes sense to keep me listed here.

Comment (dotlambda) [reply pkgs/tools/security/bitw/default.nix:23]: > Recently I decided against switching my entire setup for it

May I ask what you decided to stick with? :)

Comment (Ma27) [reply pkgs/tools/security/bitw/default.nix:23]: The very short summary is passage/age-plugin-yubikey. But... it's complicated and I wouldn't recommend that as a bit/vaultwarden replacement. Anyways, we're getting off-topic here. If you're curious, feel free to ping me :D

**Pull request closed without merging**